api/chain.yml

```diff
@@ -0,0 +1,307 @@
+        readOnly: true
+      nonce:
+        type: number
+        readOnly: true
+      from: 
+        type: string
+        readOnly: true
+      to: 
+        type: string
+      value:
+        type: number
+      gasPrice: 
+        type: number
+      gasLimit:
+        type: number
+      method: 
+        type: string
+      parameters: 
+        type: object
+      signature: 
+        type: string
+        readOnly: true
+  Tipset:
+    type: object
+    properties:
+      kind: 
+        type: string
+        enum: [tipset]
+        readOnly: true
+      id:
+        type: string
+        readOnly: true
+        example: 'zDPWYqFCyaQ9QJHr1qWZ19rFg3YRkcuXREkvpiHRJZhGN5T8SnyJ'
+      height:
+        type: number
+      blocks:
+        type: array
+        items:
+          $ref: '#/models/BlockHeader'  
+endpoints:
+  block_resource:
+    get:
+      operationId: getBlockById
+      summary: Find block by ID
+      description: Return information about the Block with the specified ID
+      tags:
+        - Blocks
+      parameters: 
+      - name: blockId
+        in: path
+        description: ID of the Block to fetch
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: Success
+          content:
+            application/json:
+              schema:
+                $ref: '#/models/Block'
+        401:
+          $ref: 'api.yml#/components/responses/UnauthorizedError'
+  executed_message_resource:
+    get:
+      operationId: getExecutedMessageById
+      summary: Find ExecutedMessage by ID
+      description: Return information about the ExecutedMessage with the specified ID. An executed Message is one that has been included in a valid block in the chain. It includes return value and status information.
+      tags:
+        - Messages
+      parameters: 
+      - name: executedMessageId
+        in: path
+        description: ID of the executed message to fetch
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: Success
+          content:
+            application/json:
+              schema:
+                $ref: '#/models/ExecutedMessage'
+        401:
+          $ref: 'api.yml#/components/responses/UnauthorizedError'
+  message_resource:
+    get:
+      operationId: getMessageById
+      summary: Find Message by ID
+      description: Return information about the Message with the specified ID
+      tags: 
+        - nessages
+      parameters: 
+      - name: messageId
+        in: path
+        description: ID of the message to fetch
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: Success
+          content:
+            application/json:
+              schema:
+                $ref: '#/models/Message'
+        401:
+          $ref: 'api.yml#/components/responses/UnauthorizedError'
+  messages_collection:
```

Comment:
You can add a GET here which returns the list of messages in a message collection (the id of which is in the header)